update/src/update.ts

```diff
@@ -127,23 +146,104 @@ export async function updateProject(dir: string) {
     }
   });
 
+  // add completion log
+  // deno-lint-ignore no-console
+  console.log(colors.green("✅ Configuration updated successfully"));
+
+  // deno-lint-ignore no-console
+  console.log(colors.cyan("🔍 Scanning for source files..."));
+
   // Update routes folder
   const project = new tsmorph.Project();
   const sfs = project.addSourceFilesAtPaths(
     path.join(dir, "**", "*.{js,jsx,ts,tsx}"),
   );
+
+  // deno-lint-ignore no-console
+  console.log(colors.cyan(`📁 Found ${sfs.length} files to process`));
+
+  if (sfs.length === 0) {
+    // deno-lint-ignore no-console
+    console.log(colors.green("🎉 Migration completed successfully!"));
+    return;
+  }
+
+  // deno-lint-ignore no-console
+  console.log(colors.yellow("🔄 Processing files..."));
+
+  // Track progress with a single counter
+  let completedFiles = 0;
+  const modifiedFilesList: string[] = [];
+
+  // process files sequentially to show proper pro
   await Promise.all(sfs.map(async (sourceFile) => {
     try {
-      return await updateFile(sourceFile);
+      const wasModified = await updateFile(sourceFile);
+      completedFiles++;
+      if (wasModified) {
+        modifiedFilesList.push(sourceFile.getFilePath());
+      }
+
+      // update progress bar
+      const progress = createProgressBar(completedFiles, sfs.length);
+      Deno.stdout.writeSync(
+        new TextEncoder().encode(`\r${colors.blue(progress)} Complete!`),
+      );
+
+      return wasModified;
     } catch (err) {
+      completedFiles++;
+
+      // update progress bar (even on failure)
+      const progress = createProgressBar(completedFiles, sfs.length);
+      Deno.stdout.writeSync(
+        new TextEncoder().encode(`\r${colors.blue(progress)} Complete!`),
+      );
```

Comment:
Nit: Move this into a `finally` block. That way we don't need to duplicate the code to clear the progress bar twice.